### DISCUSSION TITLE
함수 내부에서 선언된 변수와 블록 내부에서 선언된 변수의 차이점은 무엇인가요?

### DISCUSSION THREAD

**Discussion by lledellebell** [diving-javascript/you-dont-know-js-yet#64] [1. 스코프] [answered]
<details>
<sumary>답변</summary>
</details>

**Comment by wogus216** [comment, marked as answer]
<details>
<summary>답변</summary>



함수 내부에서 선언된 변수(함수 스코프)는 바로 var 키워드를 지칭 하기에 변수가 선언된 함수 내 어디서든 접근이 가능하다는 것을 의미

블록 내부에서 선언된 변수(블록 스코프)는 let,const 키워드를 지칭 하고 변수가 블록 내에서만 변수가 유효

let,const 는 var 와 다르게 선언만 되고 초기화가 되지 않는 TDZ(일시적 사각지대)현상도 있습니다.
</details>
